aspnetcore/blazor/test.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Test components in ASP.NET Core Blazor
+author: guardrex
+description: Learn how to test componments in Blazor apps.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 08/10/2020
+no-loc: [cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: blazor/test
+---
+# Test components in ASP.NET Core Blazor
+
+[Egil Hansen](https://egilhansen.com/)
+
+Testing is an important aspect of building stable and maintainable software.
+
+To test a Blazor component, the *Component Under Test* (CUT) is:
+
+* Rendered with relevant input for the test.
+* Depending on the type of test performed, possibly subject to interaction or modification. For example, event handlers can be triggered, such as an `onclick` event for a button.
+* Inspected for expected values.
+
+## Test approaches
+
+Two common approaches for testing Blazor components are End-to-end (E2E) testing and unit testing:
+
+* **E2E testing**: A test runner runs a Blazor app containing the CUT and automates a browser instance. The testing tool inspects and interacts with the CUT through the browser. [Selenium](https://github.com/SeleniumHQ/selenium) is an example of an E2E testing framework that can be used with Blazor apps.
+* **Unit testing**: [Unit tests](/dotnet/core/testing/) are written with a unit testing library that provides:
+  * Component rendering.
+  * Inspection of component output and state.
+  * Triggering of event handlers and life cycle methods.
+  * Assertions that component behavior is correct.
+
+[bUnit](https://github.com/egil/bUnit) is an example of a library that enables Blazor app testing.
```

Comment:
```suggestion
[bUnit](https://github.com/egil/bUnit) is an example of a library that enables Blazor component testing.
```
Using the term "Blazor app testing" might be to broad and could confuse some folks. It is certainly possible to do `RenderComponent<App>()` and render the entire app, but I do not think such app level testing should be the recommend approach.